subsys/bluetooth/host/classic/avrcp.c

```diff
@@ -3235,6 +3235,9 @@ int bt_avrcp_ct_register_notification(struct bt_avrcp_ct *ct, uint8_t tid, uint8
 	buf = avrcp_prepare_vendor_pdu(ct->avrcp, BT_AVRCP_PKT_TYPE_SINGLE, BT_AVRCP_CTYPE_NOTIFY,
 				       BT_AVRCP_PDU_ID_REGISTER_NOTIFICATION, param_len);
 	if (buf == NULL) {
+		ct->ct_notify[event_id].cb = NULL;
```

Comment:
could we move ct_notify after avrcp_prepare_vendor_pdu success?

	ct->ct_notify[event_id].cb = cb;
	ct->ct_notify[event_id].interim_received = 0;
	ct->ct_notify[event_id].tid = tid;